SwrveSDK/src/main/java/com/swrve/sdk/messaging/view/SwrveMessageView.java

```diff
@@ -1,250 +1,341 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
 /**
- * Android view representing a Swrve message with a given format. It contains
- * the SwrveInnerMessageView that will render buttons and images specified
- * on the template.
+ * Android view representing a Swrve message with a given format.
+ * It layouts its children around its center and supports show and dismiss animations.
  */
 public class SwrveMessageView extends RelativeLayout {
     protected static final String LOG_TAG = "SwrveMessagingSDK";
 
-    // Container dialog that will display the message
-    protected Dialog containerDialog;
-    // Message being displayed
-    protected SwrveMessage message;
+    // Activity that contains
+    private final Messenger messenger;
+
     // Message format chosen to display message
-    protected SwrveMessageFormat format;
-    // Indicates if it is the first time this message is shown. It will be
-    // false if the orientation is changing and the same message is going to
-    // be redisplayed.
-    protected boolean firstTime = true;
-    // Indicates if the message has yet to be drawn
-    protected boolean firstDraw = true;
-
-    // Reference to inner message view that will render buttons and images
-    protected SwrveInnerMessageView innerMessageView;
-
-    // Native android animation ID to launch when showing message
-    protected int showAnimation;
-    // Native android animation ID to launch when dismissing message
-    protected int dismissAnimation;
-
-    public SwrveMessageView(Context context) {
-        super(context);
-    }
+    protected final SwrveMessageFormat format;
 
-    public SwrveMessageView(Context context, AttributeSet attrs) {
-        super(context, attrs);
-    }
+    // Scale to fit in the device
+    protected float scale;
 
-    public SwrveMessageView(Context context, AttributeSet attrs, int defStyle) {
-        super(context, attrs, defStyle);
-    }
+    // Minimum sample size to use when loading images
+    protected int minSampleSize = 1;
 
-    public SwrveMessageView(Context context, SwrveMessage message, SwrveMessageFormat format, boolean firstTime, int rotation, int minSampleSize) throws SwrveMessageViewBuildException {
-        this(context, message, format, null, null, firstTime, rotation, minSampleSize);
-    }
+    // Bitmap cache
+    protected Set<WeakReference<Bitmap>> bitmapCache;
 
-    public SwrveMessageView(Context context, SwrveMessage message, SwrveMessageFormat format, ISwrveInstallButtonListener installButtonListener,
-                            ISwrveCustomButtonListener customButtonListener, boolean firstTime, int rotation, int minSampleSize) throws SwrveMessageViewBuildException {
+    public SwrveMessageView(Context context, Messenger messenger, SwrveMessage message, SwrveMessageFormat format, int minSampleSize) throws SwrveMessageViewBuildException {
```

Comment:
Could we move all the messenger and click listener logic to the activity class? Am not sure it belongs here.